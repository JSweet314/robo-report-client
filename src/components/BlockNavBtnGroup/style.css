.question-block-nav-group {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 2rem 0 1rem 0;
}

.question-block-nav-btn {
  font-size: 3rem;
  font-weight: bold;
  text-align: center;
  min-width: 90px;
  border-radius: 2px;
  box-shadow: 0 2px 3px 1px rgba(0, 0, 0, 0.14), 
              0 2px 6px 1px rgba(0, 0, 0, 0.12),
              0 3px 1px -2px rgba(0, 0, 0, 0.2);
  padding: 1rem;
  background-color: #e0e0e0;
  transition: background-color .3s, box-shadow .3s;
}

.question-block-nav-btn.back-btn:hover {
  background-color: #dadada;
  box-shadow: 0 3px 3px 1px rgba(0, 0, 0, 0.14),
              0 5px 10px 2px rgba(0, 0, 0, 0.12),
              0 3px 3px -3px rgba(0, 0, 0, 0.4);
  transition: background-color .5s, box-shadow .5s;
}

.question-block-nav-btn.next-btn {
  color: #fff;
  background-color: #3896d4;
  box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 
              0 1px 5px 0 rgba(0, 0, 0, 0.12), 
              0 3px 1px -2px rgba(0, 0, 0, 0.2);
  transition: background-color .5s, color .5s, box-shadow .5s;
}

.question-block-nav-btn.next-btn:hover {
  background-color: #1487d3;
  box-shadow: 0 3px 3px 1px rgba(0, 0, 0, 0.14),
              0 5px 10px 2px rgba(0, 0, 0, 0.12),
              0 3px 3px -3px rgba(0, 0, 0, 0.4);
  transition: background-color .5s, box-shadow .5s;
}

.question-block-nav-btn.next-btn:disabled {
  cursor: initial;
  color: gray;
  background-color: #ebebeb;
  box-shadow: none;
}

.question-block-nav-btn.next-btn:active {
  background: #106096;
}