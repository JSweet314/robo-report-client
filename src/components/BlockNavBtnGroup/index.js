import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const BlockNavBtnGroup = (
  { handleQuestionBlockNavigation, isNextBtnDisabled, blockIndex }
) => {

  const nextText = blockIndex > 3 ? 'Report It!' : 'Next';

  return (
    <div className='question-block-nav-group'>
      <button
        className='question-block-nav-btn back-btn'
        name='back'
        onClick={event => handleQuestionBlockNavigation(event)}
      >
        Back
      </button>
      <button
        disabled={isNextBtnDisabled && blockIndex < 4}
        className='question-block-nav-btn next-btn'
        name='next'
        onClick={event => handleQuestionBlockNavigation(event)}
      >
        { nextText }
      </button>
    </div>
  );
};

BlockNavBtnGroup.propTypes = {
  handleQuestionBlockNavigation: PropTypes.func.isRequired,
  isNextBtnDisabled: PropTypes.bool.isRequired,
  blockIndex: PropTypes.number.isRequired
};

export default BlockNavBtnGroup;