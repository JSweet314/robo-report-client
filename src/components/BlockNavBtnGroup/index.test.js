import React from 'react';
import { shallow } from 'enzyme';
import BlockNavBtnGroup from './index';

describe('BlockNavBtnGroup', () => {
  let wrapper, 
    mockHandleQuestionBlockNavigation, 
    mockIsNextBtnDisabled, 
    mockBlockIndex;

  beforeEach(() => {
    mockHandleQuestionBlockNavigation = jest.fn();
    mockIsNextBtnDisabled = true;
    mockBlockIndex = 0;
    wrapper = shallow(
      <BlockNavBtnGroup
        blockIndex={mockBlockIndex}
        handleQuestionBlockNavigation={mockHandleQuestionBlockNavigation}
        isNextBtnDisabled={mockIsNextBtnDisabled}
      />
    );
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});