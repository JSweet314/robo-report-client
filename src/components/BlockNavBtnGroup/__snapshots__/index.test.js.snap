// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockNavBtnGroup should match a snapshot 1`] = `
<div
  className="question-block-nav-group"
>
  <button
    className="question-block-nav-btn back-btn"
    name="back"
    onClick={[Function]}
  >
    Back
  </button>
  <button
    className="question-block-nav-btn next-btn"
    disabled={true}
    name="next"
    onClick={[Function]}
  >
    Next
  </button>
</div>
`;
